### PR TITLE
MySQL source resource

### PR DESCRIPTION
**Pull request by bobbyiliev**
This is mainly intended to lay the ground work for the MySQL source resource. Probably there are some missing configuration options that I'll have to add here.

### CODE REVIEW

File: docs/resources/source_mysql.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    alias = "alias_mysql_table1"
+  }
+  table {
+    name  = "shop.mysql_table2"
+    alias = "alias_mysql_table2"
+  }
+}
+
+# CREATE SOURCE schema.source_mysql
+#   FROM MYSQL CONNECTION "database"."schema"."pg_connection" (PUBLICATION 'mz_source')
```
Comment (rjobanp) [inline docs/resources/source_mysql.md:38]:
```suggestion
#   FROM MYSQL CONNECTION "database"."schema"."mysql_connection" (PUBLICATION 'mz_source')
```

File: docs/resources/source_mysql.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+- `name` (String) The identifier for the source.
+
+### Optional
+
+- `cluster_name` (String) The cluster to maintain this source.
+- `comment` (String) **Public Preview** Comment on an object in the database.
+- `database_name` (String) The identifier for the source database. Defaults to `MZ_DATABASE` environment variable if set or `materialize` if environment variable is not set.
+- `ownership_role` (String) The owernship role of the object.
+- `region` (String) The region to use for the resource connection. If not set, the default region is used.
+- `schema_name` (String) The identifier for the source schema. Defaults to `public`.
```
Comment (rjobanp) [inline docs/resources/source_mysql.md:57]: this is slightly confusing since it refers to the schema on the materialize side that holds the source object, not the schema of the upstream mysql database where the tables are. Is there a way to reword this to be more clear?

File: pkg/materialize/source_mysql.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		Source: Source{b, obj.Name, obj.SchemaName, obj.DatabaseName},
+	}
+}
+
+func (b *SourceMySQLBuilder) ClusterName(c string) *SourceMySQLBuilder {
+	b.clusterName = c
+	return b
+}
+
+func (b *SourceMySQLBuilder) Size(s string) *SourceMySQLBuilder {
```
Comment (rjobanp) [inline pkg/materialize/source_mysql.go:30]: this isn't relevant anymore, right?

File: docs/resources/source_mysql.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+- `table` (Block List) Specifies the tables to be included in the source. If not specified, all tables are included. (see [below for nested schema](#nestedblock--table))
```
Comment (rjobanp) [inline docs/resources/source_mysql.md:58]: should we also add support for the `FOR SCHEMAS` option?

File: pkg/resources/resource_source_mysql.go [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+	sourceName := d.Get("name").(string)
+	schemaName := d.Get("schema_name").(string)
+	databaseName := d.Get("database_name").(string)
+
+	metaDb, _, err := utils.GetDBClientFromMeta(meta, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	o := materialize.MaterializeObject{ObjectType: "SOURCE", Name: sourceName, SchemaName: schemaName, DatabaseName: databaseName}
+	// b := materialize.NewSource(metaDb, o)
```
Comment (rjobanp) [inline pkg/resources/resource_source_mysql.go:136]: is this relevant? should we allow updating a source object at all?

File: pkg/materialize/source_mysql.go (hunk 3 above)
Comment (bobbyiliev) [reply pkg/materialize/source_mysql.go:30]: This is used by the read func to get the size of the cluster where the source is created in:

https://github.com/MaterializeInc/terraform-provider-materialize/blob/a815ab55a8de5ba48a47d1a8cf5d003e93c7c052/pkg/materialize/source.go#L119

This was part of the migration that allowed us to remove notion of linked clusters in Terraform.

File: pkg/resources/resource_source_mysql.go (hunk 5 above)
Comment (bobbyiliev) [reply pkg/resources/resource_source_mysql.go:136]: Yes, you can update some propertites like the the name of the source, comments, ownership and etc.

I still have to work on implementing the subsources as well, eg. adding and removing subsources `ALTER SOURCE "database"."schema"."source" ADD SUBSOURCE "table_1" AS "table_alias";`

File: docs/resources/source_mysql.md (hunk 2 above)
Comment (bobbyiliev) [reply docs/resources/source_mysql.md]: Just rephrased this! Let me know what you think.

File: pkg/resources/resource_source_mysql.go (hunk 5 above)
Comment (rjobanp) [reply pkg/resources/resource_source_mysql.go:136]: Ah got it! Just so you know the `ALTER SOURCE` statements are not yet supported for MySQL.

File: docs/resources/source_mysql.md (hunk 2 above)
Comment (rjobanp) [reply docs/resources/source_mysql.md]: sounds better, thanks!

**Review by rjobanp — APPROVED** [review]
nice work!

File: pkg/resources/resource_source_mysql.go (hunk 5 above)
Comment (bobbyiliev) [reply pkg/resources/resource_source_mysql.go:138]: Ah yes, I've created a follow up issue for this: https://github.com/MaterializeInc/terraform-provider-materialize/issues/490

**Review by rjobanp — APPROVED** [review]
test_columns/ignore_columns addition lgtm
